src/main/java/com/cst438/controller/StudentController.java

```diff
@@ -2,80 +2,261 @@
-     //  remove the following line when done
-       return null;
-   }
+       if (student == null){
+           throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "user not found");
+       }
+       if (!student.getType().equals("STUDENT")){
+           throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "not authorized for this user");
+       }
 
+       List<EnrollmentDTO> schedule = new ArrayList<>();
+       List<Enrollment> enrollments = enrollmentRepository.findByYearAndSemesterOrderByCourseId(year, semester, student.getId());
+       //we are not checking for empty enrollments/ enrollments not found
+       for (Enrollment e : enrollments) {
+
+           if (!e.getStudent().getType().equals("STUDENT")) {
+               throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "user is not a student");
+           } else if (student.getId() != e.getStudent().getId()) {
+               throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "student id does not match");
+           }
+           schedule.add(
+                   new EnrollmentDTO(
+                           e.getEnrollmentId(),
+                           e.getGrade(),
+                           e.getStudent().getId(),
+                           e.getStudent().getName(),
+                           e.getStudent().getEmail(),
+                           e.getSection().getCourse().getCourseId(),
+                           e.getSection().getCourse().getTitle(),
+                           e.getSection().getSecId(),
+                           e.getSection().getSectionNo(),
+                           e.getSection().getBuilding(),
+                           e.getSection().getRoom(),
+                           e.getSection().getTimes(),
+                           e.getSection().getCourse().getCredits(),
+                           e.getSection().getTerm().getYear(),
+                           e.getSection().getTerm().getSemester()
+                   )
+           );
+       }
+       return schedule;
+   }
 
     // student adds enrollment into a section
     // user must be student
     // return EnrollmentDTO with enrollmentId generated by database
     @PostMapping("/enrollments/sections/{sectionNo}")
+    @PreAuthorize("hasAuthority('SCOPE_ROLE_STUDENT')")
     public EnrollmentDTO addCourse(
 		    @PathVariable int sectionNo,
-            @RequestParam("studentId") int studentId ) {
+            Principal principal) {
+
+        User student = userRepository.findByEmail(principal.getName());
 
-        // TODO
+        //check that user exists
+        User u = userRepository.findById(student.getId()).orElse(null);
+        if (u == null){
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "user not found");
+        }
+        if (!u.getType().equals("STUDENT")){
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "not authorized for this user");
+        }
```

Comment:
Doesn't hurt to leave it, but may be redundant.

Reply:
Yeah, I went back and added those messages just in case.  If you guys think they are just redundant, I can remove them no problem.